packages/vats/src/proposals/probeZcfBundle.js

```diff
@@ -0,0 +1,72 @@
+import { E } from '@endo/far';
+import { makeStorageNodeChild } from '@agoric/internal/src/lib-chainStorage.js';
+
+// verify that Zoe remembers the zcfBundleCap across upgrades
+export const probeZcfBundleCap = async (
+  {
+    consume: {
+      vatAdminSvc,
+      walletFactoryStartResult,
+      provisionPoolStartResult,
+      chainStorage,
+      walletBridgeManager: walletBridgeManagerP,
+      vatStore,
+    },
+  },
+  options,
+) => {
+  const { zoeRef, zcfRef, walletRef } = options.options;
+
+  const { adminNode, root: zoeRoot } = await E(vatStore).get('zoe');
+  const zoeConfigFacet = await E(zoeRoot).getZoeConfigFacet();
+
+  // STEP 1: restart WF; it'll use the newest ZCF known to Zoe ////////////////
+
+  const WALLET_STORAGE_PATH_SEGMENT = 'wallet';
+  const [walletBridgeManager, walletStorageNode, ppFacets] = await Promise.all([
+    walletBridgeManagerP,
+    makeStorageNodeChild(chainStorage, WALLET_STORAGE_PATH_SEGMENT),
+    provisionPoolStartResult,
+  ]);
+  const walletReviver = await E(ppFacets.creatorFacet).getWalletReviver();
+  const privateArgs = {
+    storageNode: walletStorageNode,
+    walletBridgeManager,
+    walletReviver,
+  };
+
+  const { adminFacet: walletAdminFacet } = await walletFactoryStartResult;
+  await E(walletAdminFacet).upgradeContract(walletRef.bundleID, privateArgs);
+
+  // STEP 2: Set the ZCF bundle ////////////////////////
+  await E(zoeConfigFacet).updateZcfBundleId(zcfRef.bundleID);
+
+  // STEP 3: Upgrade Zoe again ////////////////////////
+  // //////  See if Zoe forgets ZcfBundleCap //////////
```

Comment:
where is this seeing happening?

Reply:
so it happens after this whole proposal executes. Why put these comments after each step?

Reply:
i have no suggestions